### PR TITLE
Andy/autogen migrate

### PR DESCRIPTION
**Pull request by andrew-wm-arthur**
use the same tag generation method in the migration that we do in creating tables

The old migration just assigned sequential tags to all of the columns in the repo.

### CODE REVIEW

File: go/libraries/doltcore/rebase/rebase_tag.go [hunk 1]
```diff
@@ -73,6 +74,120 @@ func NeedsUniqueTagMigration(ctx context.Context, dEnv *env.DoltEnv) (bool, erro
 	return false, nil
 }
 
+// MigrateUniqueTags rebases the history of the repo to uniquify tags within branch histories.
```
Comment (andrew-wm-arthur) [inline go/libraries/doltcore/rebase/rebase_tag.go:77]: this is a copy/paste, it should be at top of this file

**Comment by andrew-wm-arthur** [comment]
The diff for the PR isn't great. The [second commit](https://github.com/liquidata-inc/dolt/pull/518/commits/e6a9236275fafe8eec8c290cfd727ad35f64397f) contains the change

File: go/libraries/doltcore/rebase/rebase_tag.go [hunk 2]
```diff
@@ -234,6 +351,12 @@ func replayCommitWithNewTag(ctx context.Context, root, parentRoot, rebasedParent
 			return nil, err
 		}
 
+		rebasedParentRows, err = dropValsForDeletedColumns(ctx, root.VRW().Format(), rebasedParentRows, rebasedSch, rebasedParentSch)
```
Comment (zachmu) [inline go/libraries/doltcore/rebase/rebase_tag.go:354]: Do we do this in the general case of dropping a column? We didn't use to

Comment (andrew-wm-arthur) [reply go/libraries/doltcore/rebase/rebase_tag.go:354]: We do, yes. https://github.com/liquidata-inc/dolt/pull/386

**Pull request merged**